cache.go

```diff
@@ -318,8 +432,213 @@ func (c *GoBuildCache) ToRunOption(distroKey string, opts ...GoBuildCacheOption)
 	})
 }
 
-// BazelCache sets up a cache for bazel builds.
-//
+// CargoBuildCache is a cache for Rust/Cargo build artifacts.
+// It uses sccache to speed up Rust compilation by caching build artifacts.
+type CargoBuildCache struct {
+	// Scope adds extra information to the cache key.
+	// This is useful to differentiate between different build contexts if required.
+	//
+	// This is mainly intended for internal testing purposes.
+	Scope string `json:"scope,omitempty" yaml:"scope,omitempty"`
+
+	// The cargobuild cache may be automatically injected into a build if
+	// rust is detected.
+	// Disabled explicitly turns this off.
+	Disabled bool `json:"disabled,omitempty" yaml:"disabled,omitempty"`
+}
+
+func (c *CargoBuildCache) validate() error {
+	return nil
+}
+
+type CargoBuildCacheInfo struct {
+	Platform *ocispecs.Platform
+}
+
+type CargoBuildCacheOption interface {
+	SetCargoBuildCacheOption(*CargoBuildCacheInfo)
+}
+
+type CargoBuildCacheOptionFunc func(*CargoBuildCacheInfo)
+
+func (f CargoBuildCacheOptionFunc) SetCargoBuildCacheOption(info *CargoBuildCacheInfo) {
+	f(info)
+}
+
+func WithCargoCacheConstraints(opts ...llb.ConstraintsOpt) CacheConfigOption {
+	return CacheConfigOptionFunc(func(info *CacheInfo) {
+		var c llb.Constraints
+		for _, opt := range opts {
+			opt.SetConstraintsOption(&c)
+		}
+		info.CargoBuild.Platform = c.Platform
+	})
+}
+
+const (
+	sccacheCacheDir    = "/tmp/dalec/sccache-cache"
+	sccacheBinary      = "/tmp/dalec/sccache"
+	sccacheCacheDirWin = "C:\\temp\\dalec\\sccache-cache"
+	sccacheBinaryWin   = "C:\\temp\\dalec\\sccache.exe"
+)
+
+func (c *CargoBuildCache) ToRunOption(worker llb.State, distroKey string, opts ...CargoBuildCacheOption) llb.RunOption {
+	return RunOptFunc(func(ei *llb.ExecInfo) {
+		if c.Disabled {
+			return
+		}
+
+		var info CargoBuildCacheInfo
+		for _, opt := range opts {
+			opt.SetCargoBuildCacheOption(&info)
+		}
+
+		// Ensure platform is set
+		var platform *ocispecs.Platform
+		if ei.Platform != nil {
+			platform = ei.Platform
+		} else if info.Platform != nil {
+			platform = info.Platform
+		} else {
+			p := platforms.DefaultSpec()
+			platform = &p
+		}
+
+		key := fmt.Sprintf("%s-%s-dalec-cargobuildcache", distroKey, platforms.Format(*platform))
+		if c.Scope != "" {
+			key = fmt.Sprintf("%s-%s", key, c.Scope)
+		}
+
+		// Extract distro name for needsPrecompiled check
+		distroName := distroKey
+		if idx := strings.Index(distroKey, "/"); idx != -1 {
+			distroName = distroKey[:idx]
+		}
+
+		// Map version IDs to base names for needsPrecompiled check
+		distroNameMap := map[string]string{
+			"ubuntu22.04": "jammy",
+			"ubuntu20.04": "focal",
+			"ubuntu18.04": "bionic",
+			"debian11":    "bullseye",
+		}
+		if baseName, ok := distroNameMap[distroName]; ok {
+			distroName = baseName
+		}
+
+		// Check if this distro needs precompiled sccache binary or can use package manager
+		needsPrecompiled := map[string]bool{
```

Comment:
Let's use precompiled for all the things, assuming the precompiled works and remove the swithcing.